CMSgov/easi-app
# [EASI-2727, EASI-2728] - TRB supporting documents

# EASI-2727
# EASI-2728

## Changes and Description

- Extracted document upload component from initial request form
- Added TRB documents table to Admin request home
- Added Supporting documents table and route/link from task list
- Add document remove modal on all tables
- Updated unit tests

I don't think the the TRB Admin Home/Supporting documents needs a unit test as it is just a header and the `DocumentsTable` which is already unit tested.  Similarly with the view for rendering the table from the task list

![Screenshot 2023-03-22 at 12 55 10 PM](https://user-images.githubusercontent.com/95709965/226979955-b311aebd-27c5-4b90-99ad-3694db49d548.png)

![Screenshot 2023-03-22 at 12 55 23 PM](https://user-images.githubusercontent.com/95709965/226980016-d79c5e7c-ca8f-491e-9d00-798cfff39ad6.png)

![Screenshot 2023-03-22 at 12 55 54 PM](https://user-images.githubusercontent.com/95709965/226980191-b78ad526-dce0-42ff-b9df-cd6bde1c3923.png)

![Screenshot 2023-03-22 at 12 58 01 PM](https://user-images.githubusercontent.com/95709965/226980733-183401c8-8e44-4efe-a63d-cd2570813ecc.png)

## How to test this change

- Verify existing initial request form document flow is unchanged by changes
- Create a TRB request with/without documents
- Navigate to `/trb/task-list/:id/documents
- Upload documents from the link on this page
- Verify all verbiage is reflected by location
- Remove document and verify modal
- Verify documents are rendered correctly in TRB Admin home

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
